omega/omega-connector/omega-connector-grpc/src/test/java/org/apache/servicecomb/saga/omega/connector/grpc/tcc/LoadBalanceContextBuilderTest.java

```diff
@@ -0,0 +1,123 @@
+
+  private  LoadBalanceContextBuilder tccLoadBalanceContextBuilder;
+  private  LoadBalanceContextBuilder sagaLoadBalanceContextBuilder;
+
+  @Before
+  public void setup() throws IOException {
+    // Create a server, add service, start, and register for automatic graceful shutdown.
+    grpcCleanup.register(InProcessServerBuilder.forName("localhost:8080").directExecutor().build().start());
+    grpcCleanup.register(InProcessServerBuilder.forName("localhost:8090").directExecutor().build().start());
+    when(clusterConfig.getAddresses()).thenReturn(Lists.newArrayList(addresses));
+    when(clusterConfig.getTccMessageHandler()).thenReturn(tccMessageHandler);
+    tccLoadBalanceContextBuilder =
+        new LoadBalanceContextBuilder(TransactionType.TCC, clusterConfig, serviceConfig, 30);
+    sagaLoadBalanceContextBuilder =
+        new LoadBalanceContextBuilder(TransactionType.SAGA, clusterConfig, serviceConfig, 30);
+  }
+
+  @After
+  public void teardown() {
+  }
+
+  @Test
+  public void buildTccLoadBalanceContextWithoutSsl() {
+    when(clusterConfig.isEnableSSL()).thenReturn(false);
+
+    LoadBalanceContext loadContext = tccLoadBalanceContextBuilder.build();
+    assertThat(loadContext.getPendingTaskRunner().getReconnectDelay(), is(30));
+    assertThat(loadContext.getSenders().size(), is(2));
+    assertThat(loadContext.getSenders().keySet().iterator().next(), instanceOf(TccMessageSender.class));
+    assertThat(loadContext.getSenders().values().iterator().next(), is(0l));
+    assertThat(loadContext.getChannels().size(), is(2));
+    loadContext.getSenders().keySet().iterator().next().close();
+    shutdownChannels(loadContext);
+  }
+
+  @Test
+  public void buildTccLoadBalanceContextWithSsl() {
+    when(clusterConfig.isEnableSSL()).thenReturn(true);
+    when(clusterConfig.getCert()).thenReturn(getClass().getClassLoader().getResource("client.crt").getFile());
+    when(clusterConfig.getCertChain()).thenReturn(getClass().getClassLoader().getResource("ca.crt").getFile());
+    when(clusterConfig.getKey()).thenReturn(getClass().getClassLoader().getResource("client.pem").getFile());
+    LoadBalanceContext loadContext = tccLoadBalanceContextBuilder.build();
+    assertThat(loadContext.getPendingTaskRunner().getReconnectDelay(), is(30));
+    assertThat(loadContext.getSenders().size(), is(2));
+    assertThat(loadContext.getSenders().keySet().iterator().next(), instanceOf(TccMessageSender.class));
+    assertThat(loadContext.getSenders().values().iterator().next(), is(0l));
+    assertThat(loadContext.getChannels().size(), is(2));
+    shutdownChannels(loadContext);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void throwExceptionWhenAddressIsNotExist() {
+    when(clusterConfig.getAddresses()).thenReturn(new ArrayList<String>());
+    tccLoadBalanceContextBuilder.build();
+  }
+
+  @Test
+  public void buildSagaLoadBalanceContextWithoutSsl() {
+
```

Comment:
Do we miss some test code here?

Reply:
I'll add these case when integrated with saga implement